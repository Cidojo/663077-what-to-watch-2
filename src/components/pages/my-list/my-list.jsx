import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {Header} from './../../parts/header/header.jsx';
import {Footer} from './../../parts/footer/footer.jsx';
import {Catalog} from './../../parts/catalog/catalog.jsx';
import withAuthAccess from './../../../hocs/with-auth-access/with-auth-access.jsx';
import Selectors from './../../../selectors/selectors.js';
import withActiveItem from './../../../hocs/with-active-item/with-active-item.jsx';

import {movieCardPropTypes} from './../../../global-custom-types.js';

const CatalogWrapped = withActiveItem(Catalog);

const MyList = (props) => {
  const {favoriteCards} = props;

  return (
    <div className="user-page">
      <Header
        extraClassName="user-page__head"
        title="My list"
        isWithUserBlock={true}
      />
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <CatalogWrapped
          movieCards={favoriteCards}
        />
      </section>
      <Footer/>
    </div>
  );
};

MyList.propTypes = {
  favoriteCards: PropTypes.arrayOf(movieCardPropTypes).isRequired,
};

MyList.defaultProps = {
  favoriteCards: [],
};

const mapStateToProps = (state) => ({
  favoriteCards: Selectors.getFavoriteCards(state),
});

export {MyList};

export default connect(mapStateToProps)(withAuthAccess(MyList));
