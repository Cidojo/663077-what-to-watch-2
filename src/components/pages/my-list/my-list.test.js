import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MyList} from './my-list.jsx';

configure({adapter: new Adapter()});

const props = {
  favoriteCards: []
};

it(`should render MyList component without errors`, () => {
  const tree = shallow(<MyList {...props}/>);
  expect(toJson(tree)).toMatchSnapshot();
});
