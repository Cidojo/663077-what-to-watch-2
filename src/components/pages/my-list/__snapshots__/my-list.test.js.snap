// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyList component without errors 1`] = `
<div
  className="user-page"
>
  <Header
    extraClassName="user-page__head"
    isWithUserBlock={true}
    title="My list"
  />
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <WithActiveItem
      movieCards={Array []}
    />
  </section>
  <Footer />
</div>
`;
